core/src/main/java/io/kestra/core/models/triggers/AbstractTrigger.java

```diff
@@ -38,12 +38,12 @@ abstract public class AbstractTrigger implements TriggerInterface {
 
     private String description;
 
-    @Valid
     @PluginProperty
     @Schema(
         title = "List of conditions in order to limit the flow trigger."
     )
-    protected List<Condition> conditions;
+    @Valid
+    protected List<@Valid @NotNull Condition> conditions;
```

Comment:
```suggestion
    protected List<@Valid @NotNull Condition> conditions;
```
I think you only need the annotation to the generic type not the List carrier type.
But anyway, it's OK for me as is.